api/operator/v1alpha1/vmdistributedcluster_types.go

```diff
@@ -0,0 +1,174 @@
+/*
+
+
+Licensed under the Apache License, Version 2.0 (the "License");
+you may not use this file except in compliance with the License.
+You may obtain a copy of the License at
+
+    http://www.apache.org/licenses/LICENSE-2.0
+
+Unless required by applicable law or agreed to in writing, software
+distributed under the License is distributed on an "AS IS" BASIS,
+WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+See the License for the specific language governing permissions and
+limitations under the License.
+*/
+
+package v1alpha1
+
+import (
+	"encoding/json"
+	"fmt"
+
+	corev1 "k8s.io/api/core/v1"
+	apiextensionsv1 "k8s.io/apiextensions-apiserver/pkg/apis/apiextensions/v1"
+	metav1 "k8s.io/apimachinery/pkg/apis/meta/v1"
+	"sigs.k8s.io/controller-runtime/pkg/client"
+
+	vmv1beta1 "github.com/VictoriaMetrics/operator/api/operator/v1beta1"
+)
+
+// VMDistributedClusterSpec defines the desired state of VMDistributedClusterSpec
+// +k8s:openapi-gen=true
+type VMDistributedClusterSpec struct {
+	// ParsingError contents error with context if operator was failed to parse json object from kubernetes api server
+	ParsingError string `json:"-" yaml:"-"`
+	// VMAgent points to the VMAgent object for collecting metrics from multiple VMClusters
+	VMAgent corev1.LocalObjectReference `json:"vmAgent,omitempty"`
```

Comment:
```suggestion
	VMAgent corev1.LocalObjectReference `json:"vmagent,omitempty"`
```
It's ok to use all low cases for vmagent, vmalert, like vmselect or vminsert in vmcluster.